packages/nextjs/src/middleware/middleware.ts

```diff
@@ -48,8 +48,10 @@ export async function proxyRequest(
     return NextResponse.next()
   }
 
+  const appBaseHost = request.headers.get("host");
+
   const matchBaseUrl = new URL(orySdkUrl())
-  const selfUrl = request.nextUrl.protocol + "//" + request.nextUrl.host
+  const selfUrl = request.nextUrl.protocol + "//" + (appBaseHost || request.nextUrl.host)
```

Comment:
The host would always be defined. So the fallback would never be used, right? https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Host